zebra-state/src/service/finalized_state/zebra_db.rs

```diff
@@ -96,6 +96,7 @@ impl ZebraDb {
         network: Network,
         debug_skip_format_upgrades: bool,
         column_families_in_code: impl IntoIterator<Item = String>,
+        read_only: bool,
```

Comment:
Format upgrades try to write to the database, so debug_skip_format_upgrades must be true when read_only is true.